torch/_dynamo/output_graph.py

```diff
@@ -1413,13 +1413,103 @@ def example_inputs(self) -> List[torch.Tensor]:
         return result
 
     def remove_unused_graphargs(self) -> None:
+        # NB: It's always OK to drop GraphArg for symbols that ended up being
+        # specialized.  You don't even have to make a guard for it, because
+        # ShapeEnv produce_guards operates on tracked_fakes, which never gets
+        # pruned.  That being said, you'll get marginally better generated
+        # guard code if you promote the guard into a Dynamo guard (since that
+        # allows for the guard to be done using C++ guards.)  If we get
+        # ShapeEnv guards to go into C++ guards, this will stop being a thing
+        # though!
+
         assert self.should_exit
+
         # Miniature DCE pass, but only for obviously trivial operations
+        def is_static_true(b_node: fx.node.Argument):
+            if b_node is True:
+                return True
```

Comment:
Is this possible? Can we have the args return things that are not `fx.Node`? If it is, the type of the input should be changed. Otherwise these first two ifs are not necessary

Reply:
Same applies to `is_symnode_arg`